### ISSUE TITLE
Restrict module memory and CPU usage is valid createOptions?

### ISSUE THREAD

**Issue by v-jiacz** [Azure/iotedge#3155]
-  **Memory**, **MemorySwap** and **CpuPeriod** createOptions is valid? I view **HostConfig** class in **HostConfig.cs** which not include **Memory**, **MemorySwap** and **CpuPeriod** properties.

https://docs.microsoft.com/en-us/azure/iot-edge/how-to-use-create-options

```
"createOptions": {
  "HostConfig": {
    "Memory": 268435456,
    "MemorySwap": 536870912,
    "CpuPeriod": 25000
  }
}
```

**[edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/HostConfig.cs](https://github.com/Azure/iotedge/blob/master/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/HostConfig.cs)**

```
// Copyright (c) Microsoft. All rights reserved.
namespace Microsoft.Azure.Devices.Edge.Agent.Docker.Models
{
    using System.Collections.Generic;
    using Newtonsoft.Json;
    using Newtonsoft.Json.Linq;

    public class HostConfig
    {
        [JsonProperty("Binds", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
        public IList<string> Binds { get; set; }

        [JsonProperty("LogConfig", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
        public LogConfig LogConfig { get; set; }

        [JsonProperty("Mounts", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
        public IList<Mount> Mounts { get; set; }

        [JsonProperty("NetworkMode", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
        public string NetworkMode { get; set; }

        [JsonProperty("PortBindings", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
        public IDictionary<string, IList<PortBinding>> PortBindings { get; set; }

        [JsonProperty("Privileged", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
        public bool Privileged { get; set; }

        [JsonExtensionData]
        public IDictionary<string, JToken> OtherProperties { get; set; }
    }
}
```

**Comment by ancaantochi** [comment]
@v-jiacz, 

The settings are valid, the code you refer to is for edgeAgent is using docker mode which is a dev mode disabled by default.
By default edge daemon is used which defined docker model: https://github.com/Azure/iotedge/blob/master/edgelet/docker-rs/src/models/host_config.rs 

Do you have issues with setting those properties?

**Comment by v-jiacz** [comment]
**MemorySwap** comment in line **113** and **CpuPeriod** comment in line **80**.  Are the two settings valid?

**Comment by ancaantochi** [comment]
yes, they are valid. There is a comment in the beginning of the file explaining why most of the properties are commented: https://github.com/Azure/iotedge/blob/master/edgelet/docker-rs/src/models/host_config.rs#L16

**Comment by v-jiacz** [comment]
Why the **DeviceCgroupRules**  is invalid?  Where the property is prevented?

```
{
	"HostConfig": {
		"DeviceCgroupRules": ["c 13:* rmw"],
		"Binds": ["/dev:/dev"]
	}
}
```

**Comment by ancaantochi** [comment]
It looks like it is a bug parsing this property, might be related to the type, not sure yet. I am investigating where it is and will update here.

**Comment by ancaantochi** [comment]
The issue was fixed for 1.0.10 release. 

Before we release you can try the release candidate that we published: mcr.microsoft.com/azureiotedge-agent:1.0.10-rc1
